app/src/main/java/org/oppia/app/story/StoryFragmentPresenter.kt

```diff
@@ -0,0 +1,66 @@
+package org.oppia.app.story
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import org.oppia.app.databinding.StoryChapterViewBinding
+import org.oppia.app.databinding.StoryFragmentBinding
+import org.oppia.app.model.ChapterPlayState
+import org.oppia.app.model.ChapterSummary
+import org.oppia.app.recyclerview.BindableAdapter
+import org.oppia.app.viewmodel.ViewModelProvider
+import javax.inject.Inject
+
+/** The controller for [StoryFragment] */
+class StoryFragmentPresenter @Inject constructor(
+  private val fragment: Fragment,
+  private val viewModelProvider: ViewModelProvider<StoryViewModel>
+) {
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?, storyId: String): View? {
+    val viewModel = getStoryViewModel()
+    viewModel.setStoryId(storyId)
+    val binding = StoryFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
+
+    viewModel.storyNameLiveData.observe(fragment, Observer<String> { storyName ->
+      (fragment.activity as? AppCompatActivity)?.supportActionBar?.title = storyName
+    })
+
+    viewModel.storyChapterLiveData.observe(fragment, Observer<List<ChapterSummary>> { chapterList ->
+      val completedChapters =
+        chapterList.filter { chapter -> chapter.chapterPlayState == ChapterPlayState.COMPLETED }
+          .size
+      val totalChapters = chapterList.size
+      binding.storyProgressChapterCompletedText.text =
+        "$completedChapters of $totalChapters Chapters Completed"
```

Comment:
We should be doing this via plurals and data-binding so that the string can be defined in strings.xml. All strings should be defined in strings.xml to facilitate translating the app in the future. See https://github.com/oppia/oppia-android/blob/introduce-topic-list-ui-structure/app/src/main/res/values/strings.xml#L14 and https://github.com/oppia/oppia-android/blob/introduce-topic-list-ui-structure/app/src/main/res/layout/topic_summary_view.xml#L53 for how this works.

Also: https://developer.android.com/guide/topics/resources/string-resource#Plurals and https://medium.com/@georgemount007/plurals-have-a-different-syntax-you-can-do-it-like-this-8851a706bd8f.

Reply:
Done